CoreTesting/src/au/gov/asd/tac/constellation/testing/construction/TestableGraphBuilder.java

```diff
@@ -0,0 +1,533 @@
+            
+            
+            final Camera oldCamera = VisualGraphUtilities.getCamera(gwm);
+            final BoundingBox box = new BoundingBox();
+            final Camera camera = new Camera(oldCamera);
+            BoundingBoxUtilities.recalculateFromGraph(box, gwm, true);
+            CameraUtilities.zoomToBoundingBox(camera, box);
+            VisualGraphUtilities.setCamera(gwm, camera);
+           
+        }
+    }
+    
+    public TestableGraphBuilder withAllTransactions() throws InterruptedException{
+        withLinearTransactions();    
+        withLoopedTransactions(); 
+        return this;
+    }
+    
+    public void withAllTransactions(final GraphWriteMethods gwm){
+        withLinearTransactions(gwm);
+        withLoopedTransactions(gwm);
+    }
+    
+    public TestableGraphBuilder withLoopedTransactions() throws InterruptedException {
+        WritableGraph wg = graph.getWritableGraph("addLoopedTransactions", true);
+        withLoopedTransactions(wg);    
+        wg.commit();
+        return this;
+    }
+    
+    
+    public void withLoopedTransactions(final GraphWriteMethods gwm){
+        for (final int vertex : vertexIds){
+            if (vertex % 3 == 0){
+                transactionIds.add(gwm.addTransaction(vertex, vertex, true));
+            }
+            
+            if (vertex % 6 == 0){
+                transactionIds.add(gwm.addTransaction(vertex, vertex, false));
+            }
+            
+            if (vertex % 12 == 0){
+                transactionIds.add(gwm.addTransaction(vertex, vertex, true));
+                transactionIds.add(gwm.addTransaction(vertex, vertex, false));
+            }
+        }
+    }
+    
+    public TestableGraphBuilder withLinearTransactions() throws InterruptedException{
+        final WritableGraph wg = graph.getWritableGraph("addLinearTransactions", true);
+        withLinearTransactions(wg);    
+        wg.commit();
+        return this;
+    }
+    
+    public void withLinearTransactions(final GraphWriteMethods gwm){
+        for (final int vertex : vertexIds){
+            if (vertex % 2 == 0){
+                transactionIds.add(gwm.addTransaction(vertex, 0, true));
+            } else {
+                transactionIds.add(gwm.addTransaction(vertex, 0, false));
+            }
+            
+            if (vertex == 10){
+                transactionIds.add(gwm.addTransaction(0, vertex, true));
+                transactionIds.add(gwm.addTransaction(0, vertex, false));
+                transactionIds.add(gwm.addTransaction(vertex, 0, true));
+                transactionIds.add(gwm.addTransaction(vertex, 0, false));
+            } 
+        }
+    }
+    
+    public TestableGraphBuilder withConnectionMode(final ConnectionMode mode) throws InterruptedException{
+        final WritableGraph wg = graph.getWritableGraph("setConnectionMode", true);
+        withConnectionMode(wg, mode);    
+        wg.commit();
```

Comment:
and here (and a couple of other functions below this one)

Reply:
Markin resolved as will be addressed following discussion on a previous comment